editor/editor_file_system.cpp

```diff
@@ -1625,7 +1723,7 @@ void EditorFileSystem::_save_filesystem_cache(EditorFileSystemDirectory *p_dir,
 		cache_string.append(itos(file_info->import_modified_time));
 		cache_string.append(itos(file_info->import_valid));
 		cache_string.append(file_info->import_group_file);
-		cache_string.append(String("<>").join({ file_info->script_class_name, file_info->script_class_extends, file_info->script_class_icon_path }));
+		cache_string.append(String("<>").join({ file_info->script_class_name, file_info->script_class_extends, file_info->script_class_icon_path, file_info->import_md5, String("<*>").join(file_info->import_dest_paths) }));
```

Comment:
This shouldn't be added here. You could just append at the end.

Reply:
I added the field there because the last field is special, it can contains the `::` separator. I don't think it will be possible to add more fields without breaking compatibility.
Check the line 367 in editor_file_system.cpp
```cpp
// The last section (deps) may contain the same splitter, so limit the maxsplit to 8 to get the complete deps.
Vector<String> split = l.split("::", true, 8);
```

Reply:
Maybe add it before last field? (will require some code to handle old cache)
In worst case we can bump cache file, so it gets invalidated.
It's just weird to have a list of script cache stuff and then push some random checksums. The cache is already quite messy.

Though if it requires invalidating anyway, it could be done in the future™ I guess.

Reply:
Yeah it's dumb that dependencies are just dumped at the end. It makes it impossible to properly add new cache entries. This should be changed, but not in this PR.